src/main/java/net/javadiscord/javabot/events/MessageLinkListener.java

```diff
@@ -0,0 +1,63 @@
+package net.javadiscord.javabot.events;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.GuildChannel;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.javadiscord.javabot.Bot;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MessageLinkListener extends ListenerAdapter {
+
+	private final Pattern MESSAGE_URL_PATTERN = Pattern.compile("https://((?:canary|ptb)\\.)discord.com/channels/[0-9]+/[0-9]+/[0-9]+");
+
+	@Override
+	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		if (event.getAuthor().isBot() || event.getAuthor().isSystem()) return;
+		Matcher matcher = MESSAGE_URL_PATTERN.matcher(event.getMessage().getContentRaw());
+		if (matcher.find()) {
+			var optional = parseMessageUrl(matcher.group(), event.getJDA());
+			optional.ifPresent(action -> action.queue(m -> event.getMessage().replyEmbeds(buildUrlEmbed(m)).queue()));
```

Comment:
`flatMap` could be used instead of the first `queue` in here.

Reply:
How would I obtain the sent message object, then?

Reply:
`action.queue(m -> event.getMessage().replyEmbeds(buildUrlEmbed(m)).queue());` is the same as `action.flatMap(m -> event.getMessage().replyEmbeds(buildUrlEmbed(m).queue();`. I didn't say it would be better, it is just an alternative.

Reply:
`action.flatMap(m -> event.getMessage().replyEmbeds(buildUrlEmbed(m).queue();` does not work

Reply:
I forgot a closing bracket: `action.flatMap(m -> event.getMessage().replyEmbeds(buildUrlEmbed(m)).queue();`